src/main/java/ws02/Vector.kt

```diff
@@ -0,0 +1,121 @@
+package ws02
+
+import java.lang.IllegalArgumentException
+import kotlin.math.pow
+import kotlin.math.sqrt
+
+open class Vector(initialSize : Int) : Comparable<Vector> {
+
+    private val _dimensions : MutableList<Double> = MutableList<Double>(initialSize) { _ -> 0.0 }
```

Comment:
Это тоже можно было оформить через backing property свойство, в котором был бы приватный set. Тогда бы это было более kotlin style.

Чтобы превратить все в kotlinStyle можно использовать концепцию backingField

```suggestion
    private var _dimensions : MutableList<Double> = MutableList<Double>(initialSize) { 0.0 }
    val dimensions: List<Double>
        get() = _dimensions
```

После этого надо удалить методы getValues, getValueOfDimensionInternal итд
Потому что доступ из вне будет к списку dimensions

vector.dimensions[0]

Reply:
Я как раз наоборот хотел скрыть внутреннюю реализацию хранения значений векторов) А через get/setValueOfDimension обеспечить контролируемый доступ извне)
Поэтому и появился метод getNormalizedIndex - чтобы преобразовать физический индекс измерения (обычно начинается с "1") в индекс списка.
А getInternal/setInternal возникли, когда решил изменить коллекцию с мапы на лист (когда увидел задание добавлять/удалять измерения) :)

Reply:
Да, я понял зачем это всё) В джава мире это ок) 
Ниже я накидал коммент с функциями операторами, они могли бы помочь в этом деле:)